### PR TITLE
Fix for path extracted equality comparisons

### PR DESCRIPTION
**Pull request by MISBMS**

### CODE REVIEW

File: src/main/java/io/appform/jsonrules/utils/ComparisonUtils.java [hunk 1]
```diff
@@ -86,7 +88,12 @@ static int compare(JsonNode evaluatedNode, String value) {
     public static boolean compareForEquality(ExpressionEvaluationContext context, JsonNode evaluatedNode,
             Object value) {
         final boolean nodeMissingOrNullCheck = isNodeMissingOrNull(evaluatedNode);
-        JsonNode jsonNode = mapper.valueToTree(JsonPath.read(context.getNode().toString(), String.valueOf(value)));
+        JsonNode jsonNode = MissingNode.getInstance();
+        try {
+            jsonNode = mapper.valueToTree(JsonPath.read(context.getNode().toString(), String.valueOf(value)));
+        } catch (PathNotFoundException e) {
+            // consume silently; indicates path @value doesn't exist.
```
Comment (santanusinha) [inline src/main/java/io/appform/jsonrules/utils/ComparisonUtils.java]: Is there any way to test for this? Doing this through exceptions will be really non-performant.

Comment (MISBMS) [reply src/main/java/io/appform/jsonrules/utils/ComparisonUtils.java]: Let me add a couple of test cases to ease this.
Identifying non-existence of a path through exception is better as we can open it up for a user to take a call on the eventual result. Also, PNFE comes when path doesn't exist shouldn't hinder performance. 

Further, jayway does give a way to muzzle down exceptions during path evaluations through SUPPRESS_EXCEPTION option. But, it comes at the cost of loosing out on the aforementioned flexibility of user deciding on the outcome.

**Comment by santanusinha** [comment]
How will user even do anything here? (S)he does not have a hook into this
code at all.

----
Upset the established order, and everything becomes chaos. I'm an agent of
chaos. Oh and you know the thing about chaos, it's fair.

File: src/main/java/io/appform/jsonrules/expressions/array/CollectionJsonPathBasedExpression.java [hunk 2]
```diff
@@ -62,13 +63,8 @@ protected CollectionJsonPathBasedExpression(ExpressionType type, String path, @S
...
-            JsonNode jsonNode = MissingNode.getInstance();
-            try {
-                jsonNode = mapper
-                        .valueToTree(JsonPath.read(context.getNode().toString(), String.valueOf(valuesPath)));
-            } catch (PathNotFoundException e) {
-                // consume silently; indicates path denoted by @valuesPath doesn't exist.
-            }
+            JsonNode jsonNode = mapper.valueToTree(JsonPath.using(ComparisonUtils.SUPPRESS_EXCEPTION_CONFIG)
+                    .parse(context.getNode().toString()).read(String.valueOf(valuesPath)));
             if (jsonNode == null || !jsonNode.isArray()) {
```
Comment (santanusinha) [inline src/main/java/io/appform/jsonrules/expressions/array/CollectionJsonPathBasedExpression.java]: No exactly relevant to this pull, but jsonNode == null .. check should also jave jsonNode.isNull() ?

**Pull request merged**